src/biokbase/narrative/jobs/jobmanager.py

```diff
@@ -661,33 +442,57 @@ def _cancel_job(self, job_id: str) -> None:
             del self._running_jobs[job_id]["canceling"]
 
     def retry_jobs(self, job_id_list: List[str]) -> List[dict]:
-        err_job_ids = []
-        for job_id in job_id_list:
-            if job_id is None or job_id not in self._running_jobs:
-                err_job_ids.append(job_id)
-        if len(err_job_ids) > 0:
-            # Raise exception while passing missing/bad job ids
-            raise JobException(
-                f"No jobs present with ids: {err_job_ids}", err_job_ids=err_job_ids
-            )
-
+        """
+        Returns
+        [
+            {
+                "job_id": {"state": {"job_id": job_id, "status": status, ...} ...},
+                "retry_id": {"state": {"job_id": job_id, "status": status, ...} ...}
+            },
+            ...
+            {
+                "job_id": {"state": {"job_id": job_id, "status": "does_not_exist"}},
+                "error": "does_not_exist"
+            }
+        ]
+        where the innermost dictionaries are job states from ee2 and are within the
+        job states from job.revised_state()
+        """
+        checked_jobs = self._check_job_list(job_id_list)
         try:
             retry_results = clients.get("execution_engine2").retry_jobs(
-                {"job_ids": job_id_list}
+                {"job_ids": checked_jobs["job_id_list"]}
             )
         except Exception as e:
             raise transform_job_exception(e)
+        # for each retry result, refresh the state of the retried and new jobs
+        orig_ids = [result["job_id"] for result in retry_results]
+        retry_ids = [result["retry_id"] for result in retry_results if "retry_id" in result]
+        for job_id in orig_ids:
+            self.get_job(job_id).reset_state()
```

Comment:
I don't think we need to reset the job state here -- the goal of this bit of code is to update the job state and create the output to send to the frontend. If the job is finished (e.g. in an `error` or `terminated` state), won't this remove the cached job data?

Reply:
Shoot. I may have gotten confused partially because the `job.reset_state()` method was added

Reply:
@bio-boris you mentioned something about this part

Reply:
You should be OK if you just run the usual `_construct_job_state_set`, because that should update the jobs. Just remove these two lines and all should be well.

Obviously you'll want to write some tests to verify that! 😉

Reply:
@n1mus I think the way we concluded was that we wouldn't be handling the state changes if they were changed by the api/backend (Like if after the cell state was set to terminated and the flag to use the cache was enabled, a retry on the backend wouldn't change that flag)

Reply:
Really? I can remove the flag to use the cache, but what it would do is retry then send the old/terminal state back to the FE. I thought I heard some mentions of retry changing the status

Reply:
I think we were talking about if the whole cell/parent_job had a Terminal or Non Terminal state, based on whether there were jobs in certain states. So if all jobs were finished, it would be terminal. However, if someone clicked retry, it would no longer be terminal. However, if someone called the ee2 api directly, the frontend wouldn't care.